services/api/db/views.py

```diff
@@ -1,3 +1,104 @@
+
+    def get_queryset(self):
+        user_profile =  Profile.objects.filter(user=self.request.user)
+        if not user_profile:
+            raise NotFound("User Profile not found.")
+        return user_profile
+
+    def update_profile_field(self, request, field_name):
+        try:
+            profile = self.get_queryset().first()
+            if not profile:
+                raise NotFound("Profile not found")
+            field_value = getattr(profile, field_name)
+            if request.method == 'GET':
+                return Response(field_value, status=status.HTTP_200_OK)
+            elif request.method in ['POST', 'PUT']:
+                new_field_value = request.data.get(field_name)
+                if new_field_value is None:
+                    raise ValidationError(f"{field_name.capitalize()} field is required")
+                setattr(profile, field_name, new_field_value)
+                profile.save()
+                return Response(f"{field_name.capitalize()} updated successfully", status=status.HTTP_200_OK)
+            elif request.method == 'DELETE':
+                setattr(profile, field_name, None)
+                profile.save()
+                return Response(f"{field_name.capitalize()} deleted successfully", status=status.HTTP_200_OK)
+        except Exception:
+            return Response({'error': "Something went wrong! Please try after some time"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @action(detail=False, methods=['get', 'post', 'put', 'delete'])
+    def bio(self, request):
+        return self.update_profile_field(request, 'bio')
+
+    @action(detail=False, methods=['get', 'post', 'put', 'delete'])
+    def date_of_birth(self, request):
+        return self.update_profile_field(request, 'date_of_birth')
+
+    @action(detail=False, methods=['get', 'post', 'put', 'delete'])
+    def profile_image(self, request):
+        return self.update_profile_field(request, 'profile_image_url')
+
+class OrganizationSettingsViewSet(ViewSet):
+    serializer_class = OrganizationSettingsSerializer
+    renderer_classes = [UserRenderer]
+    permission_classes = [ IsAuthenticated]  # Assuming IsAdminUser is your custom permission class
+
+    def get_object(self):
+        instance = OrganizationSettings.filter(user=self.request.user)
+        if not instance:
+            raise NotFound("Organization settings not found.")
+        return instance  # Assuming there's only one OrganizationSettings instance
+
+    def retrieve(self, request):
+        try:
+            instance = self.get_object()
+            serializer = self.serializer_class(instance)
+            return Response(serializer.data)
+        except ValidationError as e:
+            return Response({'error': e.detail}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception:
+            return Response({"error": "Something went wrong! Please try after some time"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
how to handle not found and how is the proper response and status code is shared?

Reply:
those are standard rest framework (not found, ValidationError ) error DRF returns automatically appropriate response and code. 404 for not found and 400 for validation error with correct error message but we could also provide custom error to streamline it rest of code.